Rakefile

```diff
@@ -12,6 +12,102 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 #--------------------------------------------------------------------------
+
+#
+# This rakefile consists of several tasks. This section will provide an overview of the
+# tasks and their dependencies.
+#
+# -------------------------------------------------------------------------------------------------------------------------------------------
+# | No |             Task                | Description                                                                                      |
+# |----|---------------------------------|--------------------------------------------------------------------------------------------------|
+# | 1  | regen                           | Regenerates all the SDK versions for all the services and all the profiles for all the services. |
+# |    |                                 | This task has regen_sdk_versions and regen_all_profiles as prereqs.                              |
+# |----|---------------------------------|--------------------------------------------------------------------------------------------------|
+# | 2  | regen_sdk_versions              | Regenerates all the SDK versions for all the services. At the end, this task calls               |
+# |    |                                 | regen_individual_require_files as an enhancement. This task has clean_generated as prereq.       |
+# |----|---------------------------------|--------------------------------------------------------------------------------------------------|
+# | 3  | regen_all_profiles              | Regenerates all the profiles for all the services. This task has regen_rollup_profile and        |
```

Comment:
I would indicate that this task uses a specific generator - separate from autorest, and that it has dependencies on certain json files like profiles.json

Reply:
Done